sklearn/cluster/_agglomerative.py

```diff
@@ -151,29 +151,29 @@ def ward_tree(X, *, connectivity=None, n_clusters=None, return_distance=False):
 
     Parameters
     ----------
-    X : array, shape (n_samples, n_features)
+    X : array of shape (n_samples, n_features)
         feature matrix representing n_samples samples to be clustered
 
-    connectivity : sparse matrix (optional).
+    connectivity : sparse matrix, default=None
         connectivity matrix. Defines for each sample the neighboring samples
         following a given structure of the data. The matrix is assumed to
         be symmetric and only the upper triangular half is used.
         Default is None, i.e, the Ward algorithm is unstructured.
 
-    n_clusters : int (optional)
+    n_clusters : int, default=None
         Stop early the construction of the tree at n_clusters. This is
         useful to decrease computation time if the number of clusters is
         not small compared to the number of samples. In this case, the
         complete tree is not computed, thus the 'children' output is of
         limited use, and the 'parents' output should rather be used.
         This option is valid only when specifying a connectivity matrix.
 
-    return_distance : bool (optional)
+    return_distance : bool, default=None
         If True, return the distance between the clusters.
 
     Returns
     -------
-    children : 2D array, shape (n_nodes-1, 2)
+    children : 2D array of shape (n_nodes-1, 2)
```

Comment:
Saying `2D array` is redundant because the shape already provides this information. Change by `ndarray`.

Reply:
good point! will update